parser.go

```diff
@@ -1113,7 +1170,7 @@ func (p *parser) parseSchemaObject(pkgPath, pkgName, typeName string, register b
 	if strings.HasPrefix(typeName, "[]") {
 		schemaObject.Type = &arrayType
 		itemTypeName := typeName[2:]
-		schema, ok := p.KnownIDSchema[genSchemeaObjectID(pkgName, itemTypeName)]
+		schema, ok := p.KnownIDSchema[genSchemaObjectID(pkgName, itemTypeName, p.PackageAliases)]
```

Comment:
Opt. Should we create a function that does this access for us? Seems it's repeated in enough places to justify it. Something like,

```
function (p *parser) getSchema(pkgName, typeName) (*SchemaObject, bool)
```

Then all of these could just be `schema, ok := p.getSchema(pkgName, itemTypeName)`, and when we get around to refactoring goas we could cut the new `getSchema` function out into its own package alongside the other functions related to cache access. Just trying to think ahead, feel free to punt on this for now and worry about it later.

Reply:
I think that can wait for a followup smaller PR. This one has changed quite a bit I'd like to hold off on additional changes and get it merged then we can re-evaluate.